docs/content/topics/project-developers/events.md

```diff
@@ -9,4 +9,229 @@ aliases:
   - /topics/project-developers/events.md
 ---
 
-TODO
\ No newline at end of file
+# Brigade Events
+
+Events are the lingua franca in Brigade: [Gateways] emit events and [Project]
+developers write logic to handle them.
+
+In this document, we'll look at:
+
+  * The [structure of an event]
+  * How to [configure event subscriptions] on a project
+  * How to [handle events] in a Brigade script.
+
+[Gateways]: /topics/operators/gateways
+[Project]: /topics/project-developers/projects
+[structure of an event]: #event-structure
+[configure event subscriptions]: #event-subscriptions
+[handle events]: #handling-events
+
+## Event Structure
+
+A Brigade event is defined primarily by its source and type values. These two
+must be provided every time an event is emitted into Brigade. However, there
+are plenty of other fields that event creators can harness to add additional
+context and information for utilization in handlers.
+
+The full list of fields on an Event is:
+
+  * [Source](#source)
+  * [Type](#type)
+  * [Payload](#payload)
+  * [ProjectID](#project-id)
+  * [Qualifiers](#qualifiers)
+  * [Labels](#labels)
+  * [Short Title](#short-title)
+  * [Long Title](#long-title)
+  * [Source State](#source-state)
+  * [Summary](#summary)
+
+Each field is reviewed in depth below.
+
+### Source
+
+An event's Source can be thought of as its domain. In fact, the rule of thumb
+to avoid clashes is to use a URI you control. This means leading with one's own
+domain or the URL for something else one owns, like the URL for a GitHub repo.
+As an example, events emitted from Brigade's [GitHub Gateway] use a source
+value of `brigade.sh/github`.
+
+### Type
+
+The Type of an event is used to specify increased granularity of the event's
+role/purpose as it relates to the source. For example, an event emitted by
+the GitHub Gateway with Type `push` signifies that a push event has occurred
+on a particular GitHub repo.
+
+### Payload
+
+An event Payload can be used to send free-form input that may be utilized when
+handling the event. One example would be sending the original GitHub push event
+payload on the corresponding GitHub gateway event, so that these additional
+details may be parsed in the Brigade script.
+
+### Project ID
+
+Although not normally used by general-purpose gateways, a Project ID value may
+be set on an event. In such cases, the event will _only_ be eligible for
+receipt by the project indicated by this fields' value.
```

Comment:
I _really_ like that way you've described this because it doesn't present it as undermining the subscription-based model, rather as an extension of it.

I do wonder if it's worth mentioning that this is how `brig event create --project <id>` works. That could prove instructive.